### ISSUE TITLE
readme: Add how to override the Windows default Segoe UI Emoji font

### ISSUE THREAD

**Issue by 13rac1** [eosrei/emojione-color-font#15]
The readme currently says:

> **Install on Windows**
> 
> The font installs like any other font and can be specifically selected, but the system will default to the Segoe UI Emoji font.
> 
> It can be manually selected in CSS, but making it the default is still TBD.

Is there a method to override `Segoe UI Emoji`?

**Comment by m-dango** [comment]
I mashed together a small Greasemonkey script which modifies the CSS in your browser to include the font in the body.

https://gist.github.com/mienaikage/4f487dcee7d04ba2f6d4c615812af668

**Comment by 13rac1** [comment]
@mienaikage That's a good start!

A long term solution IMO need to work without Greasemonkey. `Segoe UI Emoji` cannot be deleted because it is a protected system file. The solution on Linux is a custom fontconfig and on OSX is to change the internal postscript name of the font to override Apple Color Emoji. There has to be a method to change the default emoji font in the Windows similarly. Perhaps in the registry?

**Comment by m-dango** [comment]
I've made some progress on Windows 8.1.

Under `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts` I cleared the data for `Segoe UI Emoji` and `Segoe UI Symbol`. Under `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\FontSubstitutes` I added `Segoe UI` with data set as `Emoji One Color`

Most of the emoji are now showing correctly in Firefox, however there are issues with some characters in the Windows UI.

**Comment by 13rac1** [comment]
Interesting. What are the character issues? I haven't done any Windows specific optimizations. It may need font hinting since Windows doesn't do that itself.

**Comment by m-dango** [comment]
I'm not currently at a Windows machine so I don't have a screenshot of the effect, but some of the icons affected are the search and power icons on the top right of the start menu, and the down arrow on the bottom left. They appear as rectangles.

For reference:
http://www.techspot.com/images2/news/header/2014/04/2014-04-02.png

**Comment by 13rac1** [comment]
They will appear as rectangles if the font glyph is missing. Those characters are probably in the `Segoe UI Symbol` font. You probably shouldn't remove it from the registry.

**Comment by m-dango** [comment]
Unfortunately `Segoe UI Symbol` has a lot of emoji itself, which will be displayed instead of Emoji One if it remains present:
http://www.istartedsomething.com/uploads/emojisegoe.html

I have some free time this evening so may try and do some more investigating.

**Comment by yincrash** [comment]
I accomplished this using FontForge to merge Segoe UI Symbols font and an emoji font. [Windows' MSDN page on Font Linking](https://msdn.microsoft.com/en-us/library/bb688134.aspx) made me believe it was possible just using the registry, but it didn't work out. Other than wholesale replacing the Symbols font with an emoji font, this seemed like the only approach that worked decently.

**Comment by yincrash** [comment]
I can upload a gist of the fontforge script (it's fairly simple), but this is what it ends up looking like in Windows 7. Since 7 doesn't support color, I haven't tested if this would work with the color emojis in the newer Windows versions.
![emojionewin7](https://cloud.githubusercontent.com/assets/609567/16296970/d44a42d8-38eb-11e6-9ef3-dd4d4b495c0e.png)

**Comment by 13rac1** [comment]
Is that a screenshot from Chrome? I've actually tested this font to work in color in Windows XP in Firefox. Oh... Fontforge... that's probably why. Fontforge doesn't support the SVG table. Which is why I use both Fontfont and Fonttools in [SCFBuild](https://github.com/eosrei/scfbuild).

**Comment by yincrash** [comment]
Yeah it's from Chrome on Win 7. Do the other tools have merging abilities?

**Comment by 13rac1** [comment]
Fonttools can be used to do it, but it is just a library which is why I wrote SCFbuild. There may be merge scripts in: https://github.com/googlei18n/nototools

A concern: I will not be able to distribute a font made from merging with the Microsoft copyrighted Segoe UI Symbols font.

**Comment by yincrash** [comment]
Isn't it okay to distribute a script that merges a user's existing fonts?

edit: Essentially an 'install' script a user could run (maybe with an uninstall that would put the original symbol font back.)

**Comment by 13rac1** [comment]
Yes, the script is fine, but the font is not.

Hmm... Will Windows Resource Protection allow the file to be changed?

**Comment by yincrash** [comment]
You can copy the file out of the fonts folder, merge them. Then either copy it to the fonts directory (requiring admin privileges), or open the new merged font with the font viewer and ask the user to click the Install button. I know the install button will make it override the old unmerged font, but I'm not sure if copying into the folder will if they have different filenames (you might also have to change the font registry entry for it with that route).

**Comment by 13rac1** [comment]
Ah.. So overriding a font with the same name works? This what I do for the OSX version: https://github.com/eosrei/emojione-color-font/blob/master/scfbuild-osx.yml#L24-L25

**Comment by yincrash** [comment]
Yeah, however, if you just put in a new font with the same name without the other glyphs in the Symbols font, you will have the problem @mienaikage had where there are parts of the UI that use non-emoji glyphs from the font. I'm not positive how much of 7 uses it, but it does seem that 8.1 uses it.

**Comment by yincrash** [comment]
So I successfully got color emojis to be added to the segoe ui symbol font on Win7. I cleared out the font of everything above 0xFFFF using FontForge, then modified SCFBuild to take in the cleaned font instead of creating a new font. However, it's not perfect since some of the b/w glyphs give me tofu, and there were quite a few errors that were thrown while it was building.

![firefoxvschrome](https://cloud.githubusercontent.com/assets/609567/16543206/e848ecce-4086-11e6-96c6-4bbd85ba5726.png)

**Comment by 13rac1** [comment]
Interesting! Good progress!

**Comment by yincrash** [comment]
Let me know how this PR looks. I decided to go a much simpler route that worked very well once I figured out why pyftmerge was creating a font that Windows considered invalid. I will probably also submit a PR to https://github.com/behdad/fonttools which would make the extra python script unnecessary eventually. I've been using OSX to build and the final font works fine on Windows 7. I'll try and see if I can find a way to test it on newer versions of Windows.
